### PR TITLE
[DoctrineBridge][TwigBridge] Add PHPDoc to attributes

### PR DESCRIPTION
**Pull request by squrious**
| Q             | A
| ------------- | ---
| Branch?       | 7.1
| Bug fix?      | no
| New feature?  | yes
| Deprecations? | no
| Tickets       | Part of #51920
| License       | MIT

Add PHPDoc to attributes in `Symfony\Bridge\Doctrine` and `Symfony\Bridge\Twig`.

### CODE REVIEW

File: src/Symfony/Bridge/Doctrine/Attribute/MapEntity.php [hunk 1]
```diff
@@ -21,13 +21,49 @@
 class MapEntity extends ValueResolver
 {
     public function __construct(
+        /**
+         * The entity class.
```
Comment (stof) [inline src/Symfony/Bridge/Doctrine/Attribute/MapEntity.php]: `@var class-string|null`

File: src/Symfony/Bridge/Doctrine/Attribute/MapEntity.php [hunk 2]
```diff
@@ -21,13 +21,49 @@
...
+         */
         public ?string $class = null,
+
+        /**
+         * Specify the object manager used to retrieve the entity.
+         */
         public ?string $objectManager = null,
+
+        /**
+         * An expression to fetch the entity. The following variables are available:
```
Comment (stof) [inline src/Symfony/Bridge/Doctrine/Attribute/MapEntity.php]: this should mention that this is an expression using the ExpressionLanguage syntax, as done in the other PR.

File: src/Symfony/Bridge/Doctrine/Attribute/MapEntity.php [hunk 3]
```diff
@@ -21,13 +21,49 @@
...
+         * - `repository` will be your entity repository.
+         * - Any route wildcards are available as variables.
```
Comment (stof) [inline src/Symfony/Bridge/Doctrine/Attribute/MapEntity.php]: request attributes, not route wildcards (which is not even a name used in Symfony anywhere)

File: src/Symfony/Bridge/Doctrine/Attribute/MapEntity.php [hunk 4]
```diff
@@ -21,13 +21,49 @@
...
+         */
         public ?string $expr = null,
+
+        /**
+         * Configures the properties and values to use with the findOneBy() method.
+         * The key is the route placeholder name and the value is the Doctrine property name.
```
Comment (stof) [inline src/Symfony/Bridge/Doctrine/Attribute/MapEntity.php]: `@var array<string, string>|null`

File: src/Symfony/Bridge/Doctrine/Attribute/MapEntity.php [hunk 5]
```diff
@@ -21,13 +21,49 @@
...
+         */
         public ?array $mapping = null,
+
+        /**
+         * Configures the properties that should be used in the findOneBy() method by excluding
+         * one or more properties so that not all are used.
```
Comment (stof) [inline src/Symfony/Bridge/Doctrine/Attribute/MapEntity.php]: `@var string[]|null`

File: src/Symfony/Bridge/Doctrine/Attribute/MapEntity.php [hunk 6]
```diff
@@ -21,13 +21,49 @@
...
         public ?array $exclude = null,
+
+        /**
+         * Whether to prevent null values from being used as parameters in the query.
+         */
         public ?bool $stripNull = null,
+
+        /**
+         * If an id option is configured and matches a route parameter, then the resolver will
+         * find by the primary key.
```
Comment (stof) [inline src/Symfony/Bridge/Doctrine/Attribute/MapEntity.php]: ``@var string[]|string|null`

File: src/Symfony/Bridge/Doctrine/Validator/Constraints/UniqueEntity.php [hunk 7]
```diff
@@ -45,8 +46,13 @@ class UniqueEntity extends Constraint
     protected static $errorNames = self::ERROR_NAMES;
 
     /**
-     * @param array|string      $fields     The combination of fields that must contain unique values or a set of options
-     * @param bool|array|string $ignoreNull The combination of fields that ignore null values
+     * @param array|string      $fields           The combination of fields that must contain unique values or a set of options
+     * @param bool|array|string $ignoreNull       The combination of fields that ignore null values
```
Comment (stof) [inline src/Symfony/Bridge/Doctrine/Validator/Constraints/UniqueEntity.php]:
```suggestion
     * @param bool|string[]|string $ignoreNull       The combination of fields that ignore null values
```

File: src/Symfony/Bridge/Twig/Attribute/Template.php [hunk 8]
```diff
@@ -11,6 +11,9 @@
 
 namespace Symfony\Bridge\Twig\Attribute;
 
+/**
+ * Define the template to render in the controller.
```
Comment (stof) [inline src/Symfony/Bridge/Twig/Attribute/Template.php:15]: `$vars` should be documented as `string[]|null`

File: src/Symfony/Bridge/Doctrine/Attribute/MapEntity.php (hunk 1 above)
Comment (squrious) [reply src/Symfony/Bridge/Doctrine/Attribute/MapEntity.php]: Thanks for your comments! I made the changes here and in the others as well

**Review by GromNaN — APPROVED** [review]

**Comment by OskarStark** [comment]
rebased ✅

**Review by OskarStark — CHANGES_REQUESTED** [review]
Please use `@param` annotations

**Comment by squrious** [comment]
> Please use `@param` annotations

Do you mean replacing `@var` with `@param` in CPP or moving descriptions to the constructor doc block?

**Comment by OskarStark** [comment]
Yes please move it to the constructor doc block and use `@param`

**Review by OskarStark — APPROVED** [review]

**Review by nicolas-grekas — APPROVED** [review]

**Comment by nicolas-grekas** [comment]
Thank you @squrious.

**Pull request merged**